pywbem/cim_operations.py

```diff
@@ -755,18 +817,32 @@ def _imethodcall(self, methodname, namespace, **params):
 
         if tup_tree is None:
             return None
-
-        if tup_tree[0] == 'ERROR':
-            code = int(tup_tree[1]['CODE'])
-            if 'DESCRIPTION' in tup_tree[1]:
-                raise CIMError(code, tup_tree[1]['DESCRIPTION'])
-            raise CIMError(code, 'Error code %s' % tup_tree[1]['CODE'])
-
-        if tup_tree[0] != 'IRETURNVALUE':
-            raise ParseError('Expecting IRETURNVALUE element, got %s' \
-                             % tup_tree[0])
-
-        return tup_tree
+        if len(tup_tree) == 0:
+            return None
+        # ERROR | ...
+
+        if tup_tree[0][0] == 'ERROR':
```

Comment:
The non-pulled ops should have the name of the ERROR element still at tup_tree[0], and so should the pulled ops. Only when it is successful, do thexpulled ops have a list of IRETURNVALUE and PARAMVALUEs.